src/components/GlobalSearch/GlobalSearchComponent.tsx

```diff
@@ -0,0 +1,314 @@
+};
+
+const defaultStrings: GlobalSearchStrings = {
+  assets: 'Assets',
+  timeSeries: 'Time Series',
+  events: 'Events',
+  files: 'Files',
+  noDescription: '[No Description]',
+  emptySearch: 'Nothing found',
+  searchPlaceholder: 'Search for a resource',
+};
+
+export type RenderSearchItem = (
+  strings: GlobalSearchStrings,
+  item: SearchItem
+) => React.ReactNode;
+export type ItemCallback = (item: SearchItem) => void;
+export type RenderSearchResultParams = {
+  strings: GlobalSearchStrings;
+  isLoading: boolean;
+  searchForFilter: SearchForFilter;
+  onItemSelected: ItemCallback;
+  onItemHover: ItemCallback;
+  items: SearchItem[];
+  renderSearchItem: RenderSearchItem;
+};
+
+type GlobalSearchProps = {
+  onError?: (error: CogniteError) => void;
+  /**
+   * Sets search filters per resource.
+   * It is also used to toggle search for a resource, by setting any of the fields to null | undefined.
+   * For example this will only search for assets
+   * ```
+   * <GlobalSearch searchForFilter={{assets: {}}} />
+   * ```
+   */
+  searchForFilter?: SearchForFilter;
+  strings?: Partial<GlobalSearchStrings>;
+  onItemSelected?: ItemCallback;
+  onItemHover?: ItemCallback;
+  renderSearchResult?: (params: RenderSearchResultParams) => React.ReactNode;
+  renderSearchItem?: RenderSearchItem;
+  onSearchResults?: (results: SearchItem[]) => void;
+};
+
+function titleFromResource(strings: GlobalSearchStrings, item: SearchItem) {
+  switch (item.resourceType) {
+    case 'asset':
+      return item.resource.name;
+    case 'timeSeries':
+      return item.resource.name;
+    case 'event':
+      return item.resource.description || strings.noDescription;
+    case 'file':
+      return item.resource.name;
+    default:
+      return 'N/A';
+  }
+}
+
+function iconFromResource(item: SearchItem) {
+  switch (item.resourceType) {
+    case 'asset':
+      return <Icon type="deployment-unit" />;
+    case 'timeSeries':
+      return <Icon type="line-chart" />;
+    case 'event':
+      return <Icon type="alert" />;
+    case 'file':
+      return <Icon type="file" />;
+  }
+}
+
+function defaultRenderSearchItem(
+  strings: GlobalSearchStrings,
+  item: SearchItem
+) {
+  const title = titleFromResource(strings, item);
+  const icon = iconFromResource(item);
+  return (
+    <div style={{ padding: '4px 0px' }}>
+      <span style={{ marginRight: '6px' }}>{icon}</span>
+      {title}
+    </div>
+  );
+}
+
+function defaultRenderSearchResult({
+  strings,
+  isLoading,
+  searchForFilter,
+  onItemSelected,
+  onItemHover,
+  items,
+  renderSearchItem,
+}: RenderSearchResultParams) {
+  const itemsGroupedByResourceType = groupBy(items, item => item.resourceType);
+
+  const renderListIfNotNull = (
+    renderIfNotNull: any, // Can't use generic syntax for lambda functions in tsx files
+    title: string,
+    items: SearchItem[]
+  ) => {
+    if (renderIfNotNull != null) {
+      const itemsOrEmpty = items != null ? items : [];
```

Comment:
better to use strict comparison (`!==`) to avoid typecasting. But is it possible here to have `items === null`? Cause `items` defined as `SearchItem[]` type

Reply:
Hmm, I guess it might only be `undefined | SearchItem[]`. So i can do a `!==undefined`, however personally I like treating `null` and `undefined` equally, and this typecasting doesn't hurt performance. The rest of the code afterwards assume `itemsOrEmpty` is not null | undefined.

Reply:
Got it, but in this case (considering `items` type can be `undefined | SearchItem[]`), I would rewrite it as `const itemsOrEmpty = items || [];`. Type casting comparing with `null` just doesn't look natural to me

Reply:
`||` is kind of dangerous as it only checks for truthiness, and is based on truthy laws. I would like to use `??` however our linter craps out on it (parser error), which is the nice way of doing `items != null ? items : null`.

Reply:
Anyways doesn't matter for the array type that we assume